src/parser/tera.pest

```diff
@@ -229,6 +271,25 @@ content = @{
     text
 }
 
+// set of allowed content inside for loops
+for_content = @{
+    include_tag |
+    import_macro_tag |
+    variable_tag |
+    comment_tag |
+    set_tag |
+    set_global_tag |
+    macro_definition |
+    block |
+    content_for_if |
+    forloop |
+    break_tag |
+    continue_tag |
+    for_filter_section |
+    raw |
+    text
+}
+
```

Comment:
Yeah definitely too many changes. Only a few things are actually valid in a for loop: ifs, set/set_global, include, break, continue, raw, text, filter_section. Block, import macro,macro definition cannot be in it.

I think the current state might even be wrong as it allows some stuff that shouldn't be allowed. I'll fix it myself after merging, I'm not even entirely sure what's allowed/unallowed

Reply:
Okay. I tested a block inside a loop on master and it didn't throw an error so I left things the way they were for now.

I'm still not sure how to avoid the other new for_content types (and for_ifs etc)

Reply:
Ah, I see.

I made another update, and after limiting the types allowed in for_content, the other new types could be simplified.